src/librustc/traits/select.rs

```diff
@@ -3360,8 +3374,9 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                         );
                     tcx.mk_existential_predicates(iter)
                 });
-                let source_trait = tcx.mk_dynamic(existential_predicates, r_b);
+                let source_ty = tcx.mk_dynamic(existential_predicates, r_b);
 
+                /*
```

Comment:
So basically right now all that needs doing is put a feature gate check around this code instead of uncommenting? (in order to get upcasting to parent traits if there is only a single parent trait?)

Reply:
Yeah, basically. I'll want to feature gate `data_a.principal_def_id() == data_b.principal_def_id()` in `assemble_candidates_for_unsizing` too, but that's probably about it.